src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -452,18 +453,46 @@ void MinMdnsResolver::AdvancePendingResolverStates()
         else if (resolver->IsActiveOperationalParse())
         {
             MATTER_TRACE_SCOPE("Active operational delegate call", "MinMdnsResolver");
-            ResolvedNodeData nodeData;
+            ResolvedNodeData nodeResolvedData;
+            CHIP_ERROR err = CHIP_ERROR_UNINITIALIZED;
+
+            err = resolver->Take(nodeResolvedData);
+
+            if (mActiveResolves.HasBrowseFor(chip::Dnssd::DiscoveryType::kOperational))
+            {
+                DiscoveredNodeData nodeData;
+                if (err == CHIP_NO_ERROR)
+                {
+                    OperationalNodeBrowseData opNodeData;
+                    opNodeData.peerId     = nodeResolvedData.operationalData.peerId;
+                    opNodeData.hasZeroTTL = nodeResolvedData.operationalData.hasZeroTTL;
+                    nodeData.Set<OperationalNodeBrowseData>(opNodeData);
+                }
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                }
+                else if (mDiscoveryContext != nullptr)
+                {
+                    mDiscoveryContext->OnNodeDiscovered(nodeData);
```

Comment:
I do have to ask: why are we not just setting up `DiscoveredNodeData` in this branch?  It's not used anywhere else...

Reply:
it was kept out as per previous logic, it no longer need to be branched that way. I have moved it to `err != CHIP_NO_ERROR` branch